contracts/contracts/strategies/ConvexEthMetaStrategy.sol

```diff
@@ -21,75 +21,141 @@ contract ConvexEthMetaStrategy is InitializableAbstractStrategy {
     using StableMath for uint256;
     using SafeERC20 for IERC20;
 
-    uint256 internal constant MAX_SLIPPAGE = 1e16; // 1%, same as the Curve UI
-    address internal constant ETH_ADDRESS =
+    uint256 public constant MAX_SLIPPAGE = 1e16; // 1%, same as the Curve UI
+    address public constant ETH_ADDRESS =
         0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
-    address internal cvxDepositorAddress;
-    IRewardStaking public cvxRewardStaker;
-    uint256 internal cvxDepositorPTokenId;
-    ICurveETHPoolV1 internal curvePool;
-    IERC20 internal lpToken;
-    IERC20 internal oeth;
-    IWETH9 internal weth;
+
+    // The following slots have been deprecated with immutable variables
+    // slither-disable-next-line constable-states
+    address private _deprecated_cvxDepositorAddress;
+    // slither-disable-next-line constable-states
+    address private _deprecated_cvxRewardStaker;
+    // slither-disable-next-line constable-states
+    uint256 private _deprecated_cvxDepositorPTokenId;
+    // slither-disable-next-line constable-states
+    address private _deprecated_curvePool;
+    // slither-disable-next-line constable-states
+    address private _deprecated_lpToken;
+    // slither-disable-next-line constable-states
+    address private _deprecated_oeth;
+    // slither-disable-next-line constable-states
+    address private _deprecated_weth;
+
+    // Ordered list of pool assets
+    // slither-disable-next-line constable-states
+    uint128 private _deprecated_oethCoinIndex = 1;
+    // slither-disable-next-line constable-states
+    uint128 private _deprecated_ethCoinIndex = 0;
+
+    // New immutable variables that must be set in the constructor
+    address public immutable cvxDepositorAddress;
+    IRewardStaking public immutable cvxRewardStaker;
+    uint256 public immutable cvxDepositorPTokenId;
+    ICurveETHPoolV1 public immutable curvePool;
+    IERC20 public immutable lpToken;
+    IERC20 public immutable oeth;
+    IWETH9 public immutable weth;
+
     // Ordered list of pool assets
-    uint128 internal oethCoinIndex;
-    uint128 internal ethCoinIndex;
+    uint128 public constant oethCoinIndex = 1;
+    uint128 public constant ethCoinIndex = 0;
 
-    // used to circumvent the stack too deep issue
-    struct InitializeConfig {
-        address curvePoolAddress; //Address of the Curve pool
+    /**
+     * @dev Verifies that the caller is the Strategist.
+     */
+    modifier onlyStrategist() {
+        require(
+            msg.sender == IVault(vaultAddress).strategistAddr(),
+            "Caller is not the Strategist"
+        );
+        _;
+    }
+
+    /**
+     * @dev Checks the Metapools balances have improved and the balances
+     * have not tipped to the other side.
+     */
+    modifier improveMetapoolBalance() {
```

Comment:
Are there any possible rounding errors, where the end diffs would be 1 off on these?

Reply:
Maybe on the Curve side. We are just subtracting on our side. The strategist shouldn't be aiming for a perfect 18 decimal place balance. That will avoid adding extra on-chain logic for any rounding issues

Reply:
I was think of what would happen with this wrapping withdrawAll.

Reply:
Withdraws are proportional so the pools asset balances doesn't change.